test/test_optim.py

```diff
@@ -2693,9 +2693,9 @@ def _diff_fn(p, grad, opt_differentiable_state, opt_class, kwargs, *ignored):
     p.grad = grad
     opt_differentiable_state = {k: v.clone() for k, v in opt_differentiable_state.items()}
     opt = opt_class([p], **kwargs)
-    opt.state.update(opt_differentiable_state)
+    opt.state[p].update(opt_differentiable_state)
     opt.step()
-    return (p,) + tuple(opt_differentiable_state.values())
+    return (p,) + tuple(v for v in opt_differentiable_state.values() if v.requires_grad)
```

Comment:
Is this actually needed? Gradcheck will filter out things that don't require gradients already

Reply:
actually it didn't, lol step was returned and gradcheck detected errors with it